app/gui2/src/components/GraphEditor/GraphNode.vue

```diff
@@ -244,35 +244,40 @@ const transform = computed(() => {
 })
 
 const startEpochMs = ref(0)
-let startEvent: PointerEvent | null = null
+let draggedElement: Element | undefined
+let significantMove = false
 let startPos = Vec2.Zero
 
-// TODO[ao]: Now, the dragPointer.events are preventing `click` events on widgets if they don't
-// stop pointerup and pointerdown. Now we ensure that any widget handling click does that, but
-// instead `usePointer` should be smarter.
 const dragPointer = usePointer((pos, event, type) => {
   if (type !== 'start') {
+    if (
+      !significantMove &&
+      draggedElement &&
+      (Number(new Date()) - startEpochMs.value >= MAXIMUM_CLICK_LENGTH_MS ||
+        pos.absolute.distanceSquared(startPos) >= MAXIMUM_CLICK_DISTANCE_SQ)
+    ) {
+      // If this is move, the dragged element (node's content or selection) should capture pointer
+      // events. See the comment below.
+      draggedElement.setPointerCapture?.(event.pointerId)
+      significantMove = true
+    }
     const fullOffset = pos.absolute.sub(startPos)
     emit('dragging', fullOffset)
   }
   switch (type) {
-    case 'start': {
+    case 'start':
       startEpochMs.value = Number(new Date())
-      startEvent = event
       startPos = pos.absolute
-      event.stopImmediatePropagation()
+      // usePointer makes the currentTarget capture pointer events, what would prevent any
+      // up/click event from firing in the original target. But we don't want this until we're
+      // sure user is actually dragging the node. Therefore we set pointer capture on `target`,
+      // and restore it to `currentTarget` (vel `draggedElement`) after a significant move.
+      if (event.target instanceof Element) event.target.setPointerCapture?.(event.pointerId)
```

Comment:
could this maybe be a configuration to `usePointer`? To avoid the pointer capture. I think it's time to make `pointerCapture` accept a configuration object with some defaults set, and make `requiredButtonMask` and `predicate` a part of it, since those are rarely used.